lib/router.js

```diff
@@ -421,8 +421,9 @@ Router.prototype._handlePaths = function(rootNode, spec, scope) {
 
     // Handle paths
     // Sequence the build process with `.each` to avoid race conditions
-    // while building the tree.
-    return P.each(Object.keys(paths), function(pathPattern) {
+    // while building the tree. Also sort(), so that fixed path segments sort
+    // before patterns (by virtue of `[a-zA-Z/] < '{'`).
+    return P.each(Object.keys(paths).sort(), function(pathPattern) {
```

Comment:
Here's the new ordering: 
<img width="1036" alt="screenshot 2016-08-08 17 16 21" src="https://cloud.githubusercontent.com/assets/814754/17500639/247da5aa-5d8c-11e6-8c90-8948fe2cd6a3.png">

Not sure whether it makes more sense then the old one..